feat(validation): add EmbedExpressionValidator to check embedder IDs

bjorncs opened the pull request:
Introduce EmbedExpressionValidator to ensure all embedder IDs used in
'embed' expressions reference existing components in the model. This
prevents misconfigurations where embed expressions refer to non-existent
components.

bjorncs committed:
feat(validation): add EmbedExpressionValidator to check embedder IDs

Introduce EmbedExpressionValidator to ensure all embedder IDs used in
'embed' expressions reference existing components in the model. This
prevents misconfigurations where embed expressions refer to non-existent
components.

arnej27959 commented on config-model/src/main/java/com/yahoo/vespa/model/application/validation/EmbedExpressionValidator.java:
```diff
@@ -0,0 +1,70 @@
+                                protected void doVisit(Expression e) {
+                                    if (e instanceof EmbedExpression ee) {
+                                        ee.requestedEmbedderId().ifPresent(id -> {
+                                            var fieldName = field.getName();
```
```suggestion
                                            var fieldId = schema.name() + field.getName();
```

bjorncs replied:
Fixed - see [ba700b1](https://github.com/vespa-engine/vespa/pull/34181/commits/ba700b1cafa60a37f82aebcdeb4e11dae09f75f5)

arnej27959 reviewed (commented):
good to know the schema name as well in the message

bjorncs committed:
fix(validation): track embedder ids by schema and field

Update log messages and test assertions to include schema name for
better clarity in error reporting and debugging.
Handle fields with name same in multiple schemas.

bjorncs commented:
@arnej27959 PTAL

arnej27959 reviewed (approved):
looks great!
